internal/transport/http_util_test.go

```diff
@@ -189,68 +186,6 @@ func (s) TestDecodeMetadataHeader(t *testing.T) {
 	}
 }
 
-func (s) TestDecodeHeaderH2ErrCode(t *testing.T) {
```

Comment:
It's unfortunate to lose these test cases.  Is it possible to rewrite them as tests on a whole transport instead?

Reply:
I rewrote these tests - and added them to the `transport_test.go` file since they seem more appropriate there.